### PR TITLE
add configuration interface

### PR DESCRIPTION
**Pull request by rhaas80**
This pull request adds a configuration interface to filo, similar in functionality to the one in AXL. Specifically it adds a new function `ER_Config`:

```
/**
 * Get/set ER configuration values.
 *
 * The following configuration options can be set (type in parenthesis):
 *   * "DEBUG" (int) - if non-zero, output debug information from inside
 *     ER.
 *   * "SETSIZE" (int) - set size for ER to use.
 *   * "MPI_BUF_SIZE" (byte count [IN], int [OUT]) - MPI buffer size to chunk
 *     file transfer. Must not exceed INT_MAX.
 *   .
 * Symbolic names ER_KEY_CONFIG_FOO are defined in er.h and should
 * be used instead of the strings whenever possible to guard against typos in
 * strings.
 *
 * \result If config != NULL, then return config on success.  If config=NULL
 *         (you're querying the config) then return a new kvtree on success.
 *         NULL on any failures.
 * \param config The new configuration. If config=NULL, then return a kvtree                             *                with all the configuration values.
 *
 */
kvtree* ER_Config(
  const kvtree* config /** [IN] - kvtree of options */
);
```
as well as a new test test_config to check functionality of this interface.

In ER_Config all configuration options that affect redset are passed down to redset, even when not set by the user, ie ER's defaults for these options will override redset's if ER_Config is called (but not otherwise).

Querying options will return ER's the current value of the options. If an option was passed down to redset then the value returned is ER's copy ie it does not query redset.

Options can be configured multiple times at runtime and it is the user's responsibility ensure that this does not lead to problems.

### CODE REVIEW

**Comment by adammoody** [comment]
Looks good.  We can update the bracket style, but I'll come back to that.

**Comment by rhaas80** [comment]
> Looks good. We can update the bracket style, but I'll come back to that.

Uh I thought I got the bracket style this time. Sigh.

File: src/er.c [hunk 1]
```diff
@@ -278,6 +279,154 @@ int ER_Finalize()
...
+  }
+
+  if (kvtree_util_set_int(retval, ER_KEY_CONFIG_MPI_BUF_SIZE,
+    er_mpi_buf_size) != KVTREE_SUCCESS)
+  {
+    success = 0;
+  }
+
+  if (!success)
+    kvtree_delete(&retval);
```
Comment (adammoody) [inline src/er.c:414]:
```suggestion
  if (!success) {
    kvtree_delete(&retval);
  }
```

File: src/er.h [hunk 2]
```diff
@@ -42,6 +47,33 @@ int ER_Create_Scheme(
   int erasure_blocks          /**< [IN] - number of erasure blocks to be generated */
 );
 
+/* needs to be above doxygen comment to get association right */
+typedef struct kvtree_struct kvtree;
+/**
```
Comment (adammoody) [inline src/er.h:52]:
```suggestion
typedef struct kvtree_struct kvtree;

/**
```

File: src/er.h [hunk 3]
```diff
@@ -42,6 +47,33 @@ int ER_Create_Scheme(
...
+ *   .
+ * Symbolic names ER_KEY_CONFIG_FOO are defined in er.h and should
+ * be used instead of the strings whenever possible to guard against typos in
+ * strings.
+ *
+ * \result If config != NULL, then return config on success.  If config=NULL
+ *         (you're querying the config) then return a new kvtree on success.
+ *         NULL on any failures.
+ * \param config The new configuration. If config=NULL, then return a kvtree
+ *                with all the configuration values.
```
Comment (adammoody) [inline src/er.h:70]: Let's note that the caller is responsible for deleting any returned kvtree.

Comment (rhaas80) [reply src/er.h:71]: Added in 91b7c1a

**Comment by adammoody** [comment]
Thanks again @rhaas80

**Pull request merged**